### ISSUE TITLE
Error code 8 when using scalafmt-native CLI

### ISSUE THREAD

**Issue by AvaPL** [scalameta/scalafmt#2493]
- **Version**: 3.0.0-RC2
- **Configuration**:
```
version = 3.0.0-RC2
```

## Steps

When I run scalafmt like this:

```bash
scalafmt-native -c .scalafmt.conf --list
```

## Problem

The output that I receive:
```scala
/home/runner/work/scalafmt/scalafmt/scalafmt-cli/target/graalvm-native-image/.scalafmt.conf
Error: Process completed with exit code 8.
```

## Expectation

Scalafmt CLI doesn't exit immediately with error.

## Workaround

Downgrading to 3.0.0-RC1

**Comment by kitbellew** [comment]
have you tried running on the same codebase with regular, non native version? your exit code 8 might be a bit hard to reproduce.

**Comment by AvaPL** [comment]
Non native version works completely fine. It's only a problem with the native version. I'm using native version for CI pipelines, that's how I encountered the issue.

**Comment by kitbellew** [comment]
@poslegm do you have any ideas?

**Comment by poslegm** [comment]
No ideas only observations.

1. On my desktop it fails with another weird error:
```
./scalafmt-linux-musl  --list                                                                                                                                                                           scala3   1 
Exception in thread "main" java.nio.file.NoSuchFileException: /home/runner/work/scalafmt/scalafmt/scalafmt-cli/target/graalvm-native-image
        at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:219)
        at java.nio.file.Files.newByteChannel(Files.java:370)
        at java.nio.file.Files.newByteChannel(Files.java:421)
        at java.nio.file.Files.readAllBytes(Files.java:3205)
        at org.scalafmt.util.FileOps$.readFile(FileOps.scala:57)
        at org.scalafmt.util.FileOps$.readFile(FileOps.scala:48)
        at org.scalafmt.cli.InputMethod$FileContents.readInput(InputMethod.scala:42)
        at org.scalafmt.cli.ScalafmtCoreRunner$.unsafeHandleFile(ScalafmtCoreRunner.scala:76)
        at org.scalafmt.cli.ScalafmtCoreRunner$.handleFile(ScalafmtCoreRunner.scala:63)
        at org.scalafmt.cli.ScalafmtCoreRunner$.$anonfun$run$3(ScalafmtCoreRunner.scala:42)
        at org.scalafmt.cli.ScalafmtCoreRunner$.$anonfun$run$3$adapted(ScalafmtCoreRunner.scala:41)
        at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:563)
        at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:561)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1288)
        at scala.collection.parallel.ParIterableLike$Foreach.leaf(ParIterableLike.scala:938)
        at scala.collection.parallel.Task.$anonfun$tryLeaf$1(Tasks.scala:52)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
        at scala.util.control.Breaks$$anon$1.catchBreak(Breaks.scala:97)
        at scala.collection.parallel.Task.tryLeaf(Tasks.scala:55)
        at scala.collection.parallel.Task.tryLeaf$(Tasks.scala:49)
        at scala.collection.parallel.ParIterableLike$Foreach.tryLeaf(ParIterableLike.scala:935)
        at scala.collection.parallel.AdaptiveWorkStealingTasks$AWSTWrappedTask.compute(Tasks.scala:152)
        at scala.collection.parallel.AdaptiveWorkStealingTasks$AWSTWrappedTask.compute$(Tasks.scala:148)
        at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$AWSFJTWrappedTask.compute(Tasks.scala:304)
        at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
        at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
        at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
        at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:517)
        at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)
```
2. macos native build completely [failed](https://github.com/scalameta/scalafmt/runs/2465507733?check_suite_focus=true) for latest release.

Looks like GraalVM magic 😒 . My only guess is that [this PR](https://github.com/scalameta/scalafmt/pull/2464) can broke something important for Graal builds.

**Comment by kitbellew** [comment]
@AvaPL where do you get the native image from? i don't know how it's generated or how to obtain it.

**Comment by kitbellew** [comment]
> ```scala
> /home/runner/work/scalafmt/scalafmt/scalafmt-cli/target/graalvm-native-image/.scalafmt.conf
> ```
when you ran it, what was your current working directory and where is the `.scalafmt` file located?

**Comment by kitbellew** [comment]
> ```
> ./scalafmt-linux-musl  --list                                                                                                                                                                           scala3   1 
> Exception in thread "main" java.nio.file.NoSuchFileException: /home/runner/work/scalafmt/scalafmt/scalafmt-cli/target/graalvm-native-image
> ```
Hmm... and here it is again, this `/home/runner/work/scalafmt/scalafmt/scalafmt-cli/target/graalvm-native-image`... shouldn't be a coincidence? almost like some directory is being computed during compilation rather than runtime.

**Comment by AvaPL** [comment]
Thank you for the fix. I haven't been able to check it yet but I want to answer your questions above for the sake of completeness of this issue. I obtained the native image via steps described here https://scalameta.org/scalafmt/docs/installation.html#native-image. When I tested the problem manually via CLI `.scalafmt.conf` was located in a folder in my home directory `/home/avapl/test` (not `/home/runner/...` like in the error log).
